packages/gatsby-source-filesystem/src/gatsby-node.js

```diff
@@ -183,3 +183,14 @@ See docs here - https://www.gatsbyjs.org/packages/gatsby-source-filesystem/
 }
 
 exports.setFieldsOnGraphQLNodeType = require(`./extend-file-node`)
+
+exports.createSchemaCustomization = ({ actions }) => {
+  const { createTypes } = actions
+  const typeDefs = `
+  type File implements Node {
```

Comment:
```suggestion
  type File implements Node @infer {
```

Reply:
Explicit `@infer` is recommended so that we use inference behavior that will be standard in Gatsby v3.